packages/cli/src/tools/envinfo.ts

```diff
@@ -2,20 +2,38 @@
 import envinfo from 'envinfo';
 import {EnvironmentInfo} from '../commands/doctor/types';
 
-export default async function getEnvironmentInfo() {
-  return JSON.parse(
-    await envinfo.run(
-      {
-        Binaries: ['Node', 'Yarn', 'npm', 'Watchman'],
-        IDEs: ['Xcode', 'Android Studio'],
-        SDKs: ['iOS SDK', 'Android SDK'],
-        npmPackages: ['react', 'react-native', '@react-native-community/cli'],
-        npmGlobalPackages: ['*react-native*'],
-      },
-      {
-        json: true,
-        showNotFound: true,
-      },
-    ),
-  ) as EnvironmentInfo;
+/**
+ * Returns information about the running system.
+ * If `json === true`, or no options are passed,
+ * the return type will be an `EnvironmentInfo`.
+ * If set to `false`, it will be a `string`.
+ */
+async function getEnvironmentInfo(): Promise<EnvironmentInfo>;
+async function getEnvironmentInfo(json: boolean): Promise<string>;
```

Comment:
I'm not super thrilled with this but haven't come up with a better idea yet.

* `getEnvironmentInfo()` return type will be `EnvironmentInfo` ✔
* `getEnvironmentInfo(false)` return type will be a `string` ✔
* `getEnvironmentInfo(true)` return type will be a `string` ❌

use `getEnvironmentInfo(true)`, the real return type would be `EnvironmentInfo` but TS will incorrectly say it's a `string`.

Reply:
Got it finally right 😊